realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1092,20 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Condition that finds a distinct set of objects of a specific class. As a Realm is unordered,
+     * it is undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return the query object.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmQuery<E> distinct(String fieldName) {
+        long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.STRING);
+        this.query.distinct(columnIndices);
+        return this;
```

Comment:
On RealmQuery I think `distinct` is much preferable as the context is very clear IMO. `allDistinctObject` seems very verbose to me. On `Realm` I could see `allDistinctObjects` having a better case (Right now it is called `Realm.distinct()`) as it might not be it might not be as clear what you mean by calling it on the Realm object itself.